### ISSUE TITLE
Y24-335 - Create a new barcode for normalised XP plates for the HiC Process in LIMBER

### ISSUE THREAD

**Issue by SujitDey2022** [sanger/limber#1958]
**User story**
As a Long Read Team Lead (Danni W.) I WANT to be able to print unique label for the normalised XP plate in LIMBER SO THAT the plate can be tracked and quality issues arising from this missing information can be identified in time.
  
**Details:**
In the HiC process, the XP plate is normalised on the Beckman creating a new plate. This new normalised plate does not exist in Limber so is not tracked and a unique barcode is not created. Currently the team are using the XP barcode then annotating the barcode with "norm", this is then tapped together with the XP plate.

This requirement of unique identification is necessary from a quality perspective and applies on all HiC Library Types as the plate contents may differ and cause issues in identifying the right plate.

**Who are the primary contacts for this story**
Danni W.

**Who is the nominated tester for UAT**
Danni W.

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] I should be able to view a new page for the 'norm' plate - Example: [SQPP-65289-M](https://limber.psd.sanger.ac.uk/limber_plates/85d8d59a-7fe8-11ef-a834-0242391542d4#relatives)
- [x] I should be able to view a unique barcode for this 'norm' plate
- [x] I should be able to view parent and child relationship between the PCR-XP plate and this new 'norm' plate
- The PCR-XP plate is part of a bespoke HiC pipeline - Reference: [ SQPP-64906-K](https://limber.psd.sanger.ac.uk/limber_plates/853d281e-7bdf-11ef-bdca-0242391542d4)
- [x] I should be able to print barcodes for this new 'norm' plate
- [x] I should be able to view the pool tubes on the new 'norm' plate screen instead of the PCR-XP plate.
- [x] I should be able to view the parent and child relationship between the new 'norm' plate and the pool tube when navigating to the corresponding pools - Reference: [NT1842909I](https://limber.psd.sanger.ac.uk/limber_tubes/a25ca23a-7be4-11ef-ac6d-02423c45b5b8)

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**
[James W.] Having a barcode is higher priority v/s pipeline implementation.

**Comment by SujitDey2022** [comment]
@Danniw373 I have created a research story for PSD to look into the particular ask. Please could you confirm the details of the story as expected. 

[Here is the link](https://docs.google.com/presentation/d/1TBhBZd_D8gQwqMXiiagoRwn84NJ9hiVCR4FFPzIwg2Q/edit?usp=sharing) to our PSD refinement session of this request for reference.

Also, I have tagged this as Long Read but wanted to understand whether the HiC process in LIMBER is a Long Read pipeline?

**Comment by SujitDey2022** [comment]
I am revising the user story to include only the label printing and tracking of normalized plates in Limber. This is in line with the discussion with PSD, where it was agreed that the research needed to implement a new HiC pipeline could be postponed due to the pre-existing backlog request of integrating it into Traction LIMS.

**Comment by SujitDey2022** [comment]
@Welly373  I have update the user-story with details and acceptance criteria discussed. I have also included an example from Short Read pipeline. Please can you review and confirm if the details included are as expected.

Fyi... @stevieing @andrewsparkes

**Comment by andrewsparkes** [comment]
Currently:
`LBB Lib PCR-XP` (plate) is transferred into `LBB Lib Pool Stock` (tube) as the first step in the `Bespoke PCR MX` pipeline defined in Limber (multiplex sequencing).

The `LBB Lib PCR-XP` (plate) is the final plate in all the following pipelines in Limber:
- Bespoke PCR
- Bespoke PCR BCR
- Bespoke PCR BCR HT
- Bespoke PCR TCR
- Bespoke PCR TCR HT

The example plate given above in the story is for the Bespoke PCR pipeline. I don't really understand what HiC is or what library types that includes.

Questions:
Are you asking that the additional Norm plate be added after the XP plate at the end of **all** those Bespoke pipelines? Or maybe just Bespoke PCR?
And Is the Norm plate an _optional_ extra plate? (i.e. maybe only for the HiC process you describe, for a subset of library types) or is every `LBB Lib PCR-XP` plate always going through a Norm plate after this change?
I think we can do any of those options without too much trouble, I'm just clarifying what's required.

**Comment by Welly373** [comment]
Hi Andrew,

For HiC we would like the additional norm plate to be added after the XP plate but not for all Bespoke PCR. As you suggested could this be limited to the library type Hic?
Other people use the Bespoke PCR workflow who do not need this norm plate.


Danni Weldon
Team Lead
Long Read Sequencing
Wellcome Sanger Institute
Genome Campus
Cambridge
CB10 1DR
EXT:4875

**Comment by dasunpubudumal** [comment]
Hi @Danniw373, a couple of small questions. They are all based on whether if a robot is being used the transfer between the XP plate and the norm plate. If a robot is being used:

1. Do you need a driver file which is based on concentrations for sample transfers?
2. Is bed verification required? If it is, can you please provide robot details and bed details?

Thanks.

**Comment by SujitDey2022** [comment]
@dasunpubudumal I will let Danni reply to the 1st one, but for the other one I recall @Welly373 stating that there is no need for bed verification.

**Comment by dasunpubudumal** [comment]
Thanks @SujitDey2022

**Comment by dasunpubudumal** [comment]
Hi @Welly373, currently Limber doesn't have a format that could be exported out for a Beckman (the formats that we do have are mostly for Hamilton).

Can we get some more information on the columns you require in this `.csv` file?

Thanks.

**Comment by Welly373** [comment]
We don't need limber to talk to the Beckman this csv is uploaded manually by the user. (The actions are independent from each other.) I just mentioned it as the csv used is often uploaded to Limber but for record/audit purposes only.

**Comment by dasunpubudumal** [comment]
Hi @Welly373, 

It seems our pipeline configs after library prep (i.e., sequencing configs) cannot be filtered via library types. One way of going through this would be to have a new submission type ("Limber-Bespoke - PCR HiC - NovaSeqX paired end sequencing" in the screenshot below; we can rename it as required). 

<img width="1563" alt="image" src="https://github.com/user-attachments/assets/9d3838b3-6296-45e7-a4f0-ee92fc124ba9">

<img width="1386" alt="image" src="https://github.com/user-attachments/assets/826f0438-324e-47ee-b319-6cf713d23465">

The other way of going through this would be to have two buttons in the XP plate - one for the Pool Stock Tube and the other for the XP Norm tube. We think this solution might be a little misleading for the users. 

Out of the two solutions, we think having the new submission type would be the better choice.

We wanted to get an opinion on having a new submission type. Also, if it is favourable, we would also like the name that we would need to use for it (e.g., "Limber-Bespoke - PCR HiC - NovaSeqX paired end sequencing").

Thanks.

**Comment by Welly373** [comment]
Hi,

Sorry this went off my radar a bit. I agree the option to have a new submission works best. Could this be called “Limber-PCR-HiC”. Can I request Alice Linsdell ***@***.******@***.***> is involved with the UAT for this change as the SSR creating the submissions she will be able to test this better than myself!

Danni Weldon
Team Lead
Long Read Sequencing
Wellcome Sanger Institute
Genome Campus
Cambridge
CB10 1DR
EXT:4875

**Comment by dasunpubudumal** [comment]
There was an issue related to printing labels of pool stock tubes where it only prints the last tube. This was fixed as part of this story.

**Comment by dasunpubudumal** [comment]
Unwipped releases:

- Sequencescape: `14.51.0`
- Limber: `3.65.0`
